### ISSUE TITLE
`azd config --help` does not show help for `AZD_CONFIG_DIR`

### ISSUE THREAD

**Issue by weikanglim** [Azure/azure-dev#2302]
`azd version 1.0.0 (commit ed2244cb62e7940ec5aceb3b68ea49158272dd7e)`

<img width="1444" alt="image" src="https://github.com/Azure/azure-dev/assets/2322434/a80cd352-066b-4f57-bf5d-f837a145ebe0">

Expected:
The usage text about configuring `AZD_CONFIG_DIR` should show up.

Actual:
No usage text is present.

Since azd migrated away from using `cobra`'s default display, setting a cobra command's `Long` text and also configuring `HelpOptions` means that `Long` text is never displayed.

This [line](https://github.com/Azure/azure-dev/blob/9c0328509637a319497c274468a97ab3b9d20250/cli/azd/cmd/config.go#L81) is likely dead code currently.

Let's fix the help text here.
Let's also investigate how to more cohesively avoid these types of issues from popping up.

**Comment by vhvb1989** [comment]
@weikanglim , the `long` is still used by the web-docs.
See: https://learn.microsoft.com/en-us/azure/developer/azure-developer-cli/reference#synopsis-2

That's basically why I didn't remove the `long` field.  

The mock for `azd config` removed all the info about the `AZD_CONFIG_DIR` from the help terminal, but users can still see it when they run `azd config --docs`

**Comment by weikanglim** [comment]
@vhvb1989 Is there some work we can do to unify the two codepaths? We need to be crisp about not having something like `Long` mean: a description only shown in web docs, but unused otherwise.

**Comment by vhvb1989** [comment]
> @vhvb1989 Is there some work we can do to unify the two codepaths? We need to be crisp about not having something like `Long` mean: a description only shown in web docs, but unused otherwise.

@Austinauth worked on getting the expected output for `help` commands
I followed the requirement from: https://www.figma.com/proto/iIRRiGPuKIfaSFfMJXcYle/AZD-Design-Patterns?node-id=1301-145366&starting-point-node-id=1301%3A142321

Then, while I was implementing the changes, I asked if we wanted to have the same text from the terminal to be displayed in the web-docs.  But the answer was No.  

The way I see it, `azd --docs` is like the `see more` of each `azd --help`

**Comment by Austinauth** [comment]
It's been a while since we took a pass at the `--help` experience. I can't remember the nuances of the discussion, but I know there was some back and forth on how verbose we wanted the in-terminal help experience to be. We can revisit in the near future if needed. 

This is probably something worth talking through as a small group so we're aligned on the approach.

cc: @savannahostrowski

**Comment by savannahostrowski** [comment]
Agreed - @vhvb1989 is working on #2051 this sprint so let's chat about this

**Comment by rajeshkamal5050** [comment]
@v-xuto seems minor. Can you try putting out a PR for this?

**Comment by v-xuto** [comment]
@rajeshkamal5050 Sure. We are currently working on it.

**Comment by v-xuto** [comment]
@rajeshkamal5050 PR is ready, please review.
- PR link: https://github.com/Azure/azure-dev/pull/2726
